lib/spack/spack/cmd/__init__.py

```diff
@@ -157,15 +212,28 @@ def parse_specs(args, **kwargs):
 
     sargs = args
     if not isinstance(args, six.string_types):
-        sargs = ' '.join(spack.util.string.quote(args))
-    specs = spack.spec.parse(sargs)
-    for spec in specs:
-        if concretize:
-            spec.concretize(tests=tests)  # implies normalize
-        elif normalize:
-            spec.normalize(tests=tests)
+        sargs = ' '.join(args)
+    unquoted_flags = _UnquotedFlags.extract(sargs)
```

Comment:
I'd move this inside the `except` block (just before `if unquoted_flags:`), as it's not needed except when there's an error.